Automattic/wc-calypso-bridge
# Filtering wc-navigation root back url conditionally when referred by Calypso

As a further iteration on the work done in https://github.com/Automattic/wc-calypso-bridge/pull/634, we're moving this logic to the front-end so that we can only filter the URL conditionally, depending on the origin of the user. 

In short, this PR will alter the behavior for the "WordPress Dashboard" link via the new navigation. It will conditionally take the user back to Calypso only if that's where they originated, or retain current URL back to  `wp_admin` otherwise. 

_Notes:_
- This PR is dependent on [another PR in `wc-admin`](https://github.com/woocommerce/woocommerce-admin/pull/6505) to actually work

### Testing Instructions
- Checkout branch & following instructions in [the readme](https://github.com/Automattic/wc-calypso-bridge/blob/master/README.md) to put your store in eComm mode (including [new revised gist](https://gist.github.com/psealock/531205e2c3d37be1d8ac4d3ef4f346bc))
- Checkout branch from [companion PR](https://github.com/woocommerce/woocommerce-admin/pull/6505) in `wc-admin` and activate.
- Ensure new navigation is enabled.
- Navigate from WP Admin to any WooCommerce page, click around several times to various pages, and then hit "WordPress Dashboard" link in nav. 
- Observe that you're returned to WP Admin as expected.
- You need to simulate a referred click from Calypso. There may be a better way, but this is what I've been doing: go to `https://wordpress.com/home`, and then modify the DOM to include a link back to your Store. I used the  following script, pasted in the console: _(change out `YOUR_LOCAL_URL_HERE` for the URL to your local)_
```js
const el = document.createElement( 'div' );
el.classList.add( 'site', 'is-private' );
el.innerHTML = `<div class="site is-private">
<a
  class="site__content"
  title="afkp2.wordpress.com"
  aria-label="afkp2.wordpress.com"
  href="YOUR_LOCAL_URL_HERE/wp-admin/admin.php?page=wc-admin"
  ><div
    class="site-icon is-blank"
    style="height: 32px; width: 32px; line-height: 32px; font-size: 32px"
  >
    <svg
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 24 24"
      class="gridicon gridicons-globe"
      height="18"
      width="18"
    >
      <use
        xlink:href="/calypso/images/gridicons-84d04a83ed8c3cfc40de995e9bd32649.svg#gridicons-globe"
      ></use>
    </svg>
  </div>
  <div class="site__info">
      <div class="site__title">BACK TO LOCAL</div>
      <div class="site__domain">test.link</div>
    <span class="site__badge site__badge-private">Private</span>
  </div></a
>
</div>`;

document.querySelector( '.site-selector__sites' ).appendChild( el );
```
- Click on the link that was just created at the bottom of your site selector
![image](https://user-images.githubusercontent.com/444632/109875923-f6babf00-7c25-11eb-8337-539f31091692.png)

- You'll be directed to your local WC Admin dashboard. Click around to various Woo Pages, and then click the "WordPress Dashboard" link.
- It should redirect you back to Calypso.
